homeassistant/components/heos/services.py

```diff
@@ -26,30 +28,40 @@
 HEOS_SIGN_OUT_SCHEMA = vol.Schema({})
 
 
-def register(hass: HomeAssistant, controller: Heos):
+def register(hass: HomeAssistant):
     """Register HEOS services."""
     hass.services.async_register(
         DOMAIN,
         SERVICE_SIGN_IN,
-        functools.partial(_sign_in_handler, controller),
+        functools.partial(_sign_in_handler, hass),
         schema=HEOS_SIGN_IN_SCHEMA,
     )
     hass.services.async_register(
         DOMAIN,
         SERVICE_SIGN_OUT,
-        functools.partial(_sign_out_handler, controller),
+        functools.partial(_sign_out_handler, hass),
         schema=HEOS_SIGN_OUT_SCHEMA,
     )
```

Comment:
What does sign in and out do?

Reply:
HEOS has a weird thing where the device controller firmware needs to be authenticated to the user's account independently from the mobile app. This enables the API to return the user's favorites and presets. The only way to log in the controller is via the API and these services expose those operations.